xbout/boutdataset.py

```diff
@@ -239,6 +242,50 @@ def to_restart(self, savepath='.', nxpe=None, nype=None,
             save_mfdataset(restart_datasets, paths, compute=True)
         return
 
+    def animate_list(self, variables, animate_over='t', save_as=None, show=False, fps=10,
+                     nrows=None, ncols=None, **kwargs):
+        nvars = len(variables)
+
+        if nrows is None and ncols is None:
+            ncols = int(np.ceil(np.sqrt(nvars)))
+            nrows = int(np.ceil(nvars/ncols))
+        elif nrows is None:
+            nrows = int(np.ceil(nvars/ncols))
+        elif ncols is None:
+            ncols = int(np.ceil(nvars/nrows))
+        else:
+            assert nrows*ncols >= nvars, 'nrows*ncols not as big as nvar'
+
+        fig, axes = plt.subplots(nrows, ncols, squeeze=False)
+
+        blocks = []
+        for v, ax in zip(variables, axes.flatten()):
+            data = v.bout.data
+            ndims = len(data.dims)
+            ax.set_title(data.name)
+
+            if ndims == 2:
+                blocks.append(animate_line(data=data, ax=ax, animate_over=animate_over,
```

Comment:
This is really neat, and a nice example of the usefulness of `animatplot`'s `block` abstraction.